xwiki-platform-core/xwiki-platform-component/xwiki-platform-component-wiki/src/main/java/org/xwiki/component/wiki/internal/bridge/DefaultWikiObjectComponentManagerEventListener.java

```diff
@@ -0,0 +1,165 @@
+                    this.wikiObjectComponentManagerEventListenerProxy
+                            .registerObjectComponents(baseObject.getReference(), baseObject, componentBuilder);
+                } else if (event instanceof XObjectDeletedEvent) {
+                    this.wikiObjectComponentManagerEventListenerProxy
+                            .unregisterObjectComponents(baseObject.getReference());
+                }
+            }
+        /* If we are at application startup time, we have to instanciate every document or object that we can find
+         * in the wiki */
+        } else if (event instanceof ApplicationReadyEvent || event instanceof WikiReadyEvent) {
+            // These 2 events are created when the database is ready. We register all wiki components.
+            // Collect every WikiObjectComponentBuilder
+            this.wikiObjectComponentManagerEventListenerProxy.registerAllObjectComponents();
+        }
+    }
+
+    /**
+     * Get the {@link WikiObjectComponentBuilder} associated to the given {@link DocumentReference}.
+     * This method checks if a component builder exists for the document reference and then for the local document
+     * reference. If no builder can is found, returns null.
+     *
+     * @param xClassReference the class reference used when searching for the component builder
+     * @return the first component builder that has been found, null if no builder is found
+     */
+    private WikiObjectComponentBuilder getAssociatedComponentBuilder(DocumentReference xClassReference)
+    {
+        try {
+            // First, try to find a component that has a hint related to the DocumentReference of the XClass
+            String serializedXClassReference = this.defaultEntityReferenceSerializer.serialize(xClassReference);
+            if (this.componentManager.hasComponent(WikiObjectComponentBuilder.class, serializedXClassReference)) {
+                return this.componentManager.getInstance(WikiObjectComponentBuilder.class, serializedXClassReference);
+            } else {
+                // If no component has been found, try again with a LocalDocumentReference
+                serializedXClassReference =
+                        this.localEntityReferenceSerializer.serialize(xClassReference);
+                if (this.componentManager.hasComponent(WikiObjectComponentBuilder.class, serializedXClassReference)) {
+                    return this.componentManager.getInstance(
+                            WikiObjectComponentBuilder.class, serializedXClassReference);
+                } else {
+                    return null;
+                }
+            }
+        } catch (ComponentLookupException e) {
+            /* As we test if a component is present in the component manager before fetching it, we shouldn't get any
+             * exception at this point. */
+            this.logger.warn(String.format(
```

Comment:
If we end up here is means the component exist but failed to initialize which is a serious issue and should be logged with an error and the actual exception IMO.